test/unit/gen/csr.go

```diff
@@ -76,31 +119,62 @@ func CSR(keyAlgorithm x509.PublicKeyAlgorithm, mods ...CSRModifier) (csr []byte,
 }
 
 func SetCSRDNSNames(dnsNames ...string) CSRModifier {
-	return func(c *x509.CertificateRequest) {
+	return func(c *x509.CertificateRequest) error {
 		c.DNSNames = dnsNames
+		return nil
 	}
 }
 
 func SetCSRIPAddresses(ips ...net.IP) CSRModifier {
-	return func(c *x509.CertificateRequest) {
+	return func(c *x509.CertificateRequest) error {
 		c.IPAddresses = ips
+		return nil
+	}
+}
+
+func SetCSRIPAddressesFromStrings(ips ...string) CSRModifier {
+	return func(c *x509.CertificateRequest) error {
+		var certIPs []net.IP
+		for _, ip := range ips {
+			certIPs = append(certIPs, net.ParseIP(ip))
+		}
```

Comment:
Optional: https://pkg.go.dev/net#ParseIP returns `nil` if supplied with an invalid IP. Perhaps that should be converted to an error here? I see that the original code also ignored these parse errors so I guess this will at least be backwards compatible.

Reply:
Thanks, I updated the code & added a check